docs/utilities/feature_flags.md

```diff
@@ -1,57 +1,263 @@
+      Content: |
+        {
+              "premium_features": {
+                "default": false,
+                "rules": {
+                  "customer tier equals premium": {
+                    "when_match": true,
+                    "conditions": [
+                      {
+                        "action": "EQUALS",
+                        "key": "tier",
+                        "value": "premium"
+                      }
+                    ]
+                  }
+                }
+              },
+              "feature2": {
+                "default": true
+              }
+          }
+      ContentType: 'application/json'
+
+  ConfigDeployment:
+    Type: AWS::AppConfig::Deployment
+    Properties:
+      ApplicationId: !Ref FeatureStoreApp
+      ConfigurationProfileId: !Ref FeatureStoreConfigProfile
+      ConfigurationVersion: !Ref HostedConfigVersion
+      DeploymentStrategyId: "AppConfig.AllAtOnce"
+      EnvironmentId: !Ref FeatureStoreDevEnv
+```
+
+The `Content` parameter is a json structure of the feature flags and rules.
+
 ### IAM Permissions
 
-By default, this utility provides AWS AppConfig as a configuration store. As such, you IAM Role needs permission - `appconfig:GetConfiguration` - to fetch feature flags from AppConfig.
+Because powertools needs to fetch the configuration from the AppConfig, you need to add `appconfig:GetConfiguration`
+action to your function.
 
 ### Creating feature flags
 
-> NOTE: Explain schema, provide sample boto3 script and CFN to create one
+When using the feature flags utility powertools expects specific schema stored in your AppConfig configuration which
+incldues:
+
+* list of named features
+* default value
+* set of rules that powertools will evaluate
+
+Each rule should contain:
+
+* value if condition is met
+* list of conditions with `action`, `key` and `value` attributes.
+
+Here is small example of a premium feature with a default value `false` and one rule with a condition: if the passed
+context equals `{"tier": "premium"}` return true.
+
+```json
+{
+  "premium_feature": {
+    "default": false,
+    "rules": {
+      "customer tier equals premium": {
+        "when_match": true,
+        "conditions": [
+          {
+            "action": "EQUALS",
+            "key": "tier",
+            "value": "premium"
+          }
+        ]
+      }
+    }
+  }
+}
+```
 
 #### Rules
 
+The rules are evaluated based on conditions which have the structure:
+
+```json
+{
+  "action": "EQUALS",
+  "key": "tier",
+  "value": "premium"
+}
+```
+
+The `action` configuration can have 4 different values: `EQUALS`, `STARTSWITH`, `ENDSWITH` and `CONTAINS`. If you have
+multiple rules powertools will evaluate every rule with a logical AND.
 
+### Setup a feature flag store
+
+First setup is to setup the `AppConfigStore` based on the AppConfig parameters you have created with the previous
+CloudFormation template.
+
+```python
+
+app_config = AppConfigStore(
+    environment="FeatureStore",
```

Comment:
environment is usually dev/test/production etc.
application would be the service name
and name would be the configuration name to get